storm-core/src/jvm/org/apache/storm/daemon/supervisor/Supervisor.java

```diff
@@ -17,135 +17,541 @@
+            for (String topoId : topoIds) {
+                Integer recordedVersion = -1;
+                Integer version = stormClusterState.assignmentVersion(topoId, callback);
+                if (localAssignmentVersion.containsKey(topoId) && localAssignmentVersion.get(topoId) != null) {
+                    recordedVersion = (Integer) localAssignmentVersion.get(topoId).get(IStateStorage.VERSION);
+                }
+                if (version == null) {
+                    // ignore
+                } else if (version == recordedVersion) {
+                    updateAssignmentVersion.put(topoId, localAssignmentVersion.get(topoId));
+                } else {
+                    //TODO change this so we return an object not a map with magic keys
+                    Map<String, Object> assignmentVersion = (Map<String, Object>) stormClusterState.assignmentInfoWithVersion(topoId, callback);
+                    updateAssignmentVersion.put(topoId, assignmentVersion);
+                }
+            }
+            return updateAssignmentVersion;
+        }
+        
+        protected Map<String, List<ProfileRequest>> getProfileActions(IStormClusterState stormClusterState, List<String> stormIds) throws Exception {
+            Map<String, List<ProfileRequest>> ret = new HashMap<String, List<ProfileRequest>>();
+            for (String stormId : stormIds) {
+                List<ProfileRequest> profileRequests = stormClusterState.getTopologyProfileRequests(stormId);
+                ret.put(stormId, profileRequests);
+            }
+            return ret;
+        }
+        
+        protected Map<Integer, LocalAssignment> readAssignments(Map<String, Map<String, Object>> assignmentsSnapshot,
+                String assignmentId, AtomicInteger retries) {
+            try {
+                Map<Integer, LocalAssignment> portLA = new HashMap<Integer, LocalAssignment>();
+                for (Map.Entry<String, Map<String, Object>> assignEntry : assignmentsSnapshot.entrySet()) {
+                    String topoId = assignEntry.getKey();
+                    Assignment assignment = (Assignment) assignEntry.getValue().get(IStateStorage.DATA);
+
+                    Map<Integer, LocalAssignment> portTasks = readMyExecutors(topoId, assignmentId, assignment);
+
+                    for (Map.Entry<Integer, LocalAssignment> entry : portTasks.entrySet()) {
+
+                        Integer port = entry.getKey();
+
+                        LocalAssignment la = entry.getValue();
+
+                        if (!portLA.containsKey(port)) {
+                            portLA.put(port, la);
+                        } else {
+                            throw new RuntimeException("Should not have multiple topologys assigned to one port");
+                        }
+                    }
+                }
+                retries.set(0);
+                return portLA;
+            } catch (RuntimeException e) {
+                if (retries.get() > 2) {
```

Comment:
This was hard coded before and I just moved the code over.  I also don't see a lot of value in making it configurable at this point.